src/modules/roc_fec/reader.cpp

```diff
@@ -231,50 +245,40 @@ void Reader::try_repair_() {
             continue;
         }
 
-        packet::PacketPtr pp = new (packet_pool_) packet::Packet(packet_pool_);
+        packet::PacketPtr pp = parse_repaired_packet_(buffer);
         if (!pp) {
-            roc_log(LogError, "fec reader: can't allocate packet");
             continue;
         }
 
-        if (!parser_.parse(*pp, buffer)) {
-            roc_log(LogDebug, "fec reader: can't parse repaired packet");
-            continue;
-        }
-
-        pp->set_data(buffer);
-
-        if (!check_packet_(pp)) {
-            roc_log(LogDebug, "fec reader: dropping unexpected repaired packet");
-            continue;
-        }
-
-        pp->add_flags(packet::Packet::FlagRestored);
-
         source_block_[n] = pp;
     }
 
     decoder_.end();
     can_repair_ = false;
 }
 
-bool Reader::check_packet_(const packet::PacketPtr& pp) {
-    roc_panic_if_not(has_source_);
+packet::PacketPtr Reader::parse_repaired_packet_(core::Slice<uint8_t> buffer) {
```

Comment:
Pass by reference instead of pass by value?

Reply:
Done.